### PR TITLE
fix: pull plan limit from DB when updating portal app

### PR DESCRIPTION
**Pull request by commoddity**
fixes an issue where the update portal app method expected the plan type by pulling the value from the DB instead of taking it as a update struct field.

### CODE REVIEW

File: postgres-driver/portal_app.go [hunk 1]
```diff
@@ -360,20 +360,35 @@ func (pg *PostgresDriver) UpdatePortalApp(ctx context.Context, update types.Upda
...
+						return fmt.Errorf(errPayPlanDoesntExist.Error(), update.PlanType)
+					default:
+						return err
+					}
+				}
+
+				updateApp.DailyLimit = newInt4(dailyLimit.Int32, false)
+			}
 		}
-		if update.DailyLimit != 0 {
```
[the remark this answers is not in the document]
Comment (commoddity) [reply postgres-driver/portal_app.go:371]: The custom limit field is separate and applies only to enterprise plans (it must be set for enterprise plans and may not be set for others; this is validated inside PUB).

File: postgres-driver/portal_app_test.go [hunk 2]
```diff
@@ -357,6 +366,13 @@ func (ts *PGDriverTestSuite) Test_UpdatePortalApp() {
 			},
 			err: nil,
 		},
+		{
+			name: "Should fail if an invalid plan type provided",
```
Comment (adshmh) [inline postgres-driver/portal_app_test.go:370]: Should we add a test case to cover custom daily limit here?

Comment (commoddity) [reply postgres-driver/portal_app_test.go:370]: This is tested in this test case: https://github.com/pokt-foundation/portal-db/blob/update-app-plan-type-from-db/postgres-driver/portal_app_test.go#L358

**Comment by commoddity** [comment]
> Thank you for the PR. Left a few comments to check up on.

Thank you. Addressed all the comment; I don't believe there are any blocked.

**Review by adshmh — APPROVED** [review]

**Pull request merged**